src/App.js

```diff
@@ -99,22 +100,41 @@ class App extends Component {
           .collection('users')
           .doc(user.uid)
           .onSnapshot(
-            (snapshot) => {
-              const data = snapshot.data();
+            async (snapshot) => {
 
-              if (!snapshot.exists || !data) {
+              let userData;
+              if (!snapshot.exists) {
+                const newUser = await createUserProfileDocument(user.uid, "student")
+
+                const getData = await newUser.get()
+                userData = getData.data();
+
+              }
+              else
+                userData = snapshot.data();
+
+              if (!userData) {
                 if (this.userDocumentSnapshotListener) {
                   this.userDocumentSnapshotListener();
                 }
                 this.resetState();
                 return;
-              }
+               }
+
+
+              // if (!snapshot.exists || !data) {
+              //   if (this.userDocumentSnapshotListener) {
+              //     this.userDocumentSnapshotListener();
+              //   }
+              //   this.resetState();
+              //   return;
+              // }
 
               this.setState({
                 ready: true,
                 user: user,
-                userData: data,
-                access: data.role,
+                userData: userData,
+                access: userData.role,
```

Comment:
Looks Good, but I think it's better if we've some ternary conditions for these two states, for example when we dont assign any role. I think it might happens when it executes line no `114`.  Correct me, if i'm wrong anywhere.